include/rocksdb/cloud/cloud_file_system.h

```diff
@@ -530,6 +531,27 @@ class CloudFileSystem : public FileSystem {
   virtual IOStatus CopyLocalFileToDest(const std::string& local_name,
                                        const std::string& cloud_name) = 0;
 
+  // Returns CloudManifest file name for a given db.
+  virtual std::string CloudManifestFile(const std::string& dbname) = 0;
+
+  virtual CloudManifest* GetCloudManifest() = 0;
+
+  // TODO(wei): this function is used to temporarily support open db and switch
+  // cookie. Change it to use the cookie from options once migration complete.
+  virtual IOStatus CreateCloudManifest(const std::string& local_dbname,
+                                       const std::string& cookie) = 0;
+
+  virtual IOStatus LoadCloudManifest(const std::string& local_dbname,
+                                     bool read_only) = 0;
+
+  virtual IOStatus UploadCloudManifest(const std::string& local_dbname,
+                               const std::string& cookie) const = 0;
+
+  // Prepare a local directory for use as a clone of the cloud storage
+  virtual IOStatus SanitizeLocalDirectory(const DBOptions& options,
+                                          const std::string& local_name,
+                                          bool read_only) = 0;
```

Comment:
Are these changes unrelated?

Reply:
This is related to cleaning up the APIs so that we can implement DisaggFileSystem in Rockset.